src/main.rs

```diff
@@ -67,7 +67,14 @@ fn validate_simulation() {
 
 /// Simulate the rabbit population
 fn simulate(starting_rabbits: i128) -> i32 {
-    todo!()
+    if starting_rabbits == 1 {
+        return 0;
+    }
+    if starting_rabbits % 2 == 1 {
+        return simulate(starting_rabbits * 3 + 1) + 1;
+    } else {
+        return simulate(starting_rabbits / 2) + 1;
+    }
```

Comment:
Took me a second to figure out what was going on here :P

Reply:
This is what 3007 does to people :slightly_smiling_face:

Reply:
Oh, don't worry, there will be lots of room for inspiration from 3007 :P